### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by JakobBD** [pik-piam/flodym#10]
I made this one large issue. To me, this contains

- [x] installation instructions
- [ ] A brief overview of the main classes and functions
- [x] use examples 
- [x] Maybe some sort of API reference

**Comment by JakobBD** [comment]
Great work on this.
- Can we somehow include the Readme function in the RTD? 
- The NamedDimArray class documentation is central to using sodym. It should appear quite prominently when explaining the design ideas and usage.

**Comment by JakobBD** [comment]
We could look into adapting the ODYM tutorials as examples. No need to re-invent the wheel...

**Comment by JakobBD** [comment]
How do we want to showcase all the different ways to use sodym? Here's what I mean in particular: 

- DataReader: 
  - use an existing one  
  - code your own one
  - input values directly
- Data plotting: 
  - export to df and use plotly 
  - use ArrayPlotter
- MFASystem: 
  - use from_definition
  - pre-make all inputs (flows, stocks)
- flow naming
  - standard
  - custom function

To introduce them all, we could... 

- Option 1: Just vary the way one-by-one introduce them in the existing examples. But the examples already differ in what kind of system they compute, so it could be confusing if things are done differently all of a sudden. 
- Option 2: 
  - I'd like to directly contrast the different options for each part of the library in one example. Like "You could do it like A, but also like B"
  - The current examples to me seem more like use cases. Often, they don't show new functionality, just new applications. This is different from the above. Maybe we can separate it.

**Comment by SallyDa** [comment]
I agree with option 2 and I would make separate examples for these comparisons, with file names specifying what the example is illustrating, e.g. data_reader_example, mfa_initialisation_example etc.

I also think it's good to have the use_case examples (i.e. the ODYM examples) as well, as this then shows how everything is put together and can be used in specific applications.
